### ISSUE TITLE
Error Running Drizzle-kit Migrations Through Proxy

### ISSUE THREAD

**Issue by tsatsujnr139** [TimoWilhelm/local-neon-http-proxy#6]
I have no idea what has changed because this used to work flawlessly but now I get the error below when using this with drizzle to run my migrations. (pnpm drizzle-kit migrate).  Running queries using drizzle orm is not affected. This is issue is only faced when running migrations

I'm using drizzle-orm 0.33.0 & drizzle-kit 0.24.0. 

Any help would be much appreaciated.

![CleanShot 2024-10-12 at 15 26 43@2x](https://github.com/user-attachments/assets/1e8cf1f0-52d4-4579-8a81-9e7944dc39b4)

**Comment by TimoWilhelm** [comment]
Are you using the [container images pubished on the GitHub feed](https://github.com/TimoWilhelm/local-neon-http-proxy/pkgs/container/local-neon-http-proxy) or cloning the repo locally and which version of [`@neondatabase/serverless`](https://www.npmjs.com/package/@neondatabase/serverless) are you using?

Can you share the code you use to construct the neon and drizzle clients for running the migration?

I see that there is a WebSocket URL in the error log. It should only connect using `http` urls.

**Comment by TimoWilhelm** [comment]
Closing for now.

**Comment by gadicc** [comment]
I also experienced this, posting for others who got stuck.

**TL;DR; Just ensure you have the `pg` npm package installed too.**

Here's the issue.  If you look at the full log output, you'll see this line (which was unfortunately just one line above from where OP screenshotted):

```
Using '@neondatabase/serverless' driver for database querying
```

When your *app* uses that driver, it will use the `fetchEndpoint` override as described in the README here.  But *drizzle-kit* doesn't know anything about that.  And while we could go that route, it's enough to simply `pnpm add -D pg` (or yarn, npm, etc), and the next run works great.  You'll also see at the top:

```
Using 'pg' driver for database querying
```

which confirms its using the direct driver.  Enjoy everyone :)  And thanks @TimoWilhelm for this awesome project 🙏
